nexus/lib/qmcpack.py

```diff
@@ -525,6 +557,36 @@ def write_prep(self):
                         fobj.close()
                     #end if
                 #end for
+                grand_canonical_twist_average = 'nelecs_at_twist' in self
+                if grand_canonical_twist_average:
+                    for itwist, qi in enumerate(input.inputs):
+                        elecs = self.nelecs_at_twist[itwist]
+                        # step 1: resize particlesets
+                        nup = elecs[0]
+                        ndn = elecs[1]
+                        qi.get('u').set(size=nup)
+                        qi.get('d').set(size=ndn)
+                        # step 2: resize determinants
+                        dset = qi.get('determinantset')
+                        sdet = dset.slaterdeterminant  # hard-code single det
+                        spo_size_map = {}
+                        for det, nelec in zip(sdet.determinants, elecs):
```

Comment:
`sdet.determinants` is dict-like and does not guarantee ordering.  The match for up/down needs to be done differently.

Reply:
Assuming each `<determinant>` has a `group` attribute, this matching could be done.
Please see next push.

Reply:
Looks OK now